scipy/stats/tests/test_axis_nan_policy.py

```diff
@@ -699,13 +706,14 @@ def small_sample_generator(n_dims):
 
             except ValueError:
                 # confirm that the arrays truly are not broadcastable
-                assert not _check_arrays_broadcastable(samples, axis)
+                assert not _check_arrays_broadcastable(samples,
+                                                       None if paired else axis)
```

Comment:
Once `entropy` passed that part of the test, I ran into a new problem here. 
- `entropy` samples `pk` and `qk` are paired - they need to be broadcastable to the same shapes *without* ignoring `axis`.
- For some `pk` and `qk`, `entropy` raised an error because the shapes weren't broadcastable, but `_check_arrays_broadcastable` thought they were broadcastable because it was ignoring `axis`.
- The fix is to make `axis=None` (which means don't ignore any axes, not "ravel") for paired-sample functions.